### ISSUE TITLE
Eraser shortcut

### ISSUE THREAD

**Issue by Perangelot** [Cimbali/pympress#235]
**Is your feature request related to a problem? Please describe.**
When drawing with my tablet on the slides, it often happens that I want to erase certain lines. Simply undoing the last command won't do it. I have not found a way to do that and there does not seem to be a key defined for it.

**Describe the solution you'd like**
Create a shortcut for the erase function which is only called while the key the function is bound to is pressed. This way, I could bind one of my tablet buttons to that key in order to easily erase things.

**Describe alternatives you've considered**
The best alternative of course would be tablet support, but I think this is not worth the effort.

**Comment by Perangelot** [comment]
Thanks for the fast and friendly reply! I'm afraid `0` does not open the eraser. In the configuration file, I cannot find a shortcut for it (just for `undo_scribble`). I have installed Pympress on Ubuntu using the command on you suggested.

**Comment by Cimbali** [comment]
There are no donations, see #216 for a discussion on that topic. In particular, maintaining pympress is a hobby and I intend to keep it that way − no matter how dire the need for new features is, sorry.  
It’s not as bad as it sounds though, new features arrive regularly and anyone can contribute with a PR or a suggestion (as you’re doing). And that’s more in the spirit of how I approach this project.

I already started looking into this. Holding the shortcut is doable, though not in the conventional way for actions. The drawback is that it seems impossible to handle 2 distinct cases (so we should pick one):
- keys are only modifiers, e.g. `<Alt>` or `<Ctrl>` − this seems more natural to me but leaves less choice in the key to use
- keys are not modifiers, e.g. normal letters, numbers, F-keys, etc.

**Comment by Perangelot** [comment]
Thanks for your answer! I very much understand and appreciate the way you go about things.

I am not quite sure if I have understood you correctly; are things like "CTRL-e"  possible if keys are only modifiers, or do you mean by _only_  that "CTRL" is one, but "CTRL-x" is none for any x? I'm specifically asking about "CTRL-e" because the openboard-Eraser is typically bound to "CTRL-e", too. This way, the eraser would work in both programs with the same button.

**Comment by Cimbali** [comment]
CTRL-e would be a normal shortcut. Just CTRL would be a modifier.

**Comment by Perangelot** [comment]
I understand. Then I would opt for the normal shortcut version, simply because most people who scribble on slides use a graphical pen anyhow (and there you can assign a shortcut to a button). This would also guarantee compatibility with OpenBoard. Those who you a mouse could then bind the eraser to a simple shortcut such as Ctrl-SPC.

**Comment by Cimbali** [comment]
So both are implemented, either holding just a modifier (implicitly means modifier + click) or a normal shortcut (so you can set `<ctrl>e`). You can also choose to stay in highlight mode by unticking the option in the Presentation > Highlight mode menu.

You can test by installing the current master.

@Perangelot since you’re a highlight-tool “power user” you could also look through #68 see if some suggestions sound interesting to you.

**Comment by Perangelot** [comment]
Thanks for the effort! I have pulled the latest changes and tried it out. Staying in highlight mode works perfectly! I still have not quite wrapped my head around how the eraser works, though: If I have set `highlight-use-pen::eraser = <ctrl>e` in the configuration file, pressing <ctrl>e selects the eraser - even if you let go of it.  I think this is what you just call a "shortcut".

The behavior I am looking for is this: As long as you hold <ctrl>e, the eraser is selected, and as soon as you let go of it, the last highlighting tool you worked with is selected again. Is this behavior possible at the moment? Or is it only possible with modifiers? If so, how can I configure it?

Sorry for the questions, but I have just started using github and do not know python :(
